src/SourceBuild/tarball/content/repos/Directory.Build.targets

```diff
@@ -487,6 +487,12 @@
 
     <ItemGroup>
       <SourceBuiltNupkgFiles Include="$(SourceBuiltPackagesPath)extractArtifacts/**/*.nupkg" />
+      <!-- don't copy some packages that we specifically want to override for now due to bad reference package generation                                                  -->
+      <!-- relevant issues: https://github.com/dotnet/runtime/issues/44646, https://github.com/dotnet/runtime/issues/45183, https://github.com/dotnet/runtime/issues/45417 -->
+      <!-- these should be fixed for 6.0 proper                                                                                                                            -->
+      <PrebuiltPackagesToUse Include="$(PrebuiltPackagesPath)/*.nupkg" />
+      <SourceBuiltNupkgFilesToSkip Include="@(PrebuiltPackagesToUse -> '$(SourceBuiltPackagesPath)extractArtifacts/artifacts/%(Filename)%(Extension)')" />
+      <SourceBuiltNupkgFiles Condition="'@(_BuiltIntermediatePackages)' != '' and $([System.String]::Copy(%(_BuiltIntermediatePackages.Identity)).Contains('source-build-reference-packages'))" Remove="@(SourceBuiltNupkgFilesToSkip)" />
```

Comment:
Wouldn't it be better to move `'@(_BuiltIntermediatePackages)' != ''` to the ItemGroup condition?  Why bother performing any of this logic?

Reply:
Changed as suggested, thanks.